icu4j/maven-build/maven-icu4j/create-fullLocaleNames.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# This command lists all the resource files in a single directory that we want to match on.
+# Then we filter the locale resource files, based on the file name, using the same match pattern
+# used in the corresponding target in the Ant build.xml configuration.
+# Then, we store the locale names in a file in that directory, which ICUResourceBundle.java might look for.
+#
+# Input argument $1 = directory containing containing *.res files
+ls $1/*.res\
+ | ruby -lane 'puts "basename "+$F[0]' \
```

Comment:
optional: might be nice to use something more common / lower-level than ruby?

Reply:
Perl was designed to be a superset of awk, sed, etc. in order to replace them.  Ruby was originally designed to be a replacement for Perl.  I find Ruby to be way more readable, and true to form, Ruby is indeed an easy drop-in replacement for Perl in one-liner contexts.

I think Ruby is quite common, just like Python. :-) Certainly more common than the other tools I listed above.

Reply:
> I think Ruby is quite common, just like Python. :-)

Let's agree to disagree :-)
https://www.tiobe.com/tiobe-index/
Python: 1st place, 16.66%, and going up
Ruby, 17th place, 0.81%, and going down

;-)